src/controllers/authenticationController.ts

```diff
@@ -105,6 +106,70 @@ export default class AuthenticationController {
       .send(metadata);
   }
 
+  /**
+   * Returns data about the current user session.
+   */
+  public getSessionState(req: express.Request, res: express.Response): void {
+    if (req.headers && req.headers.authorization) {
+      const authorization = req.headers.authorization as string;
+      const parts = authorization.split(" ");
+      if (parts.length === 2) {
+        const scheme = parts[0];
+        const token = parts[1];
+
+        if (/^Bearer$/i.test(scheme)) {
+          this.sessionStorage
+            .get(token)
+            .then((errorOrUser: Either<Error, User>) => {
+              errorOrUser.fold(
+                () => {
+                  this.sessionStorage.refresh(token).then(
+                    (errorOrNewToken: Either<Error, string>) => {
+                      errorOrNewToken.fold(
+                        // error while refreshing the token.
+                        err => {
+                          res.status(500).json({
+                            error: err
```

Comment:
it's not totally clear what's the expected format for returned errors.
here we return an Error() here while at line 146 we return a string. 
maybe you mean return err.message ?

Reply:
this mismatch are hard to catch as the return values in res.json are untyped